modules/Microsoft.SignalRService/webPubSub/deploy.bicep

```
@@ -0,0 +1,188 @@
+@description('Optional. The location for all Web PubSub resources.')
+param location string = resourceGroup().location
+
+@description('Required. The name of the Web PubSub resource.')
+param name string
+
+@description('Optional. Configuration Details for private endpoints. For security reasons, it is recommended to use private endpoints whenever possible.')
+param privateEndpoints array = []
+
+@allowed([
+  ''
+  'CanNotDelete'
+  'ReadOnly'
+])
+@description('Optional. Specify the type of lock.')
+param lock string = ''
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+param roleAssignments array = []
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
+@description('Optional. The unit count of the resource. 1 by default.')
+param capacity int = 1
+
+@allowed([
+  'Free_F1'
+  'Standard_S1'
+])
+@description('Optional. Pricing tier of App Configuration.')
+param sku string = 'Standard_S1'
+
+@description('Optional. Enables system assigned managed identity on the resource.')
+param systemAssignedIdentity bool = false
+
+@description('Optional. The ID(s) to assign to the resource.')
+param userAssignedIdentities object = {}
+
+@description('Optional. When set as true, connection with AuthType=aad won\'t work.')
+param disableAadAuth bool = false
+
+@description('Optional. Disables all authentication methods other than AAD authentication. For security reasons, this value should be set to `true`.')
+param disableLocalAuth bool = true
+
+@description('Optional. Control permission for data plane traffic coming from public networks while private endpoint is enabled. For security reasons, this value should be set to `Disabled`.')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param publicNetworkAccess string = 'Enabled'
```

Same proposal done to Redis Cache review
```suggestion
@description('Optional. Whether or not public network access is allowed for this resource. For security reasons it should be disabled. If not specified, it will be disabled by default if private endpoints are set.')
@allowed([
  ''
  'Enabled'
  'Disabled'
])
param publicNetworkAccess string = ''
```